scapy/arch/pcapdnet.py

```diff
@@ -516,7 +516,19 @@ def get_if_raw_hwaddr(iff):
 
         def get_if_raw_addr(ifname):
             i = dnet.intf()
-            return i.get(ifname)["addr"].data
+
+            try:
+                if_data = i.get(ifname)
+            except OSError, e:
+                if e.message == 'Device not configured':
```

Comment:
Maybe it's safer/better, if possible, to check the value of `e.errno`? Also, can you name your variable `exc`, for example, rather than `e`?